### ISSUE TITLE
Click log out button it doesn't clear session in Keycloak (IdP)

### ISSUE THREAD

**Issue by deaw0003** [temporalio/ui#711]
**Describe the bug**
I found when I click "Logout" button in temporal ui. Session in keycloak (IdP) isn't clear. While I test logout with other application that write in nodejs. When I click logout from application, it clear session in keycloak (IdP) automatically.

**To Reproduce**

1. Log in with username and password that created in keycloak (IdP).
2. Click logout button.
3. Go to keycloak management console to check session of user that I login.
4. Found session of user still login here.

**Expected behavior**
When I click "Logout" button in temporal ui, I need it to clear session in keycloak (IdP) automatically like other application.

**Screenshots**
![image](https://user-images.githubusercontent.com/21300935/179939275-b520fab7-c5f3-44d4-abf4-b7a3e493c255.png)
![image](https://user-images.githubusercontent.com/21300935/179939539-72fb38f3-7775-486a-b499-2c539897c70d.png)

**Desktop (please complete the following information):**
- OS: Windows 10 Version 21H1
- Browser: chrome Version 103.0.5060.114 (Official Build) (64-bit) 

**Server**
- OS: Ubuntu 20.04.4 LTS
- Temporal UI version: 2.2.3
- Run Temporal on docker container

**Comment by deaw0003** [comment]
Hi, Are there any update for this.

**Comment by Aman-Godara-RIA** [comment]
Came here through a google search, I am facing the same issue with Next.js & react.js in Keycloak IDP.
The reason for this behavior in my case was that the `logout()` method was just clearing cookies and was not making an api call to the IDP to log the user out.

**Comment by feedmeapples** [comment]
> Any release plan for this issue ?

right, afaik @robholland looked into this and the solution was not straightforward?

**Comment by feedmeapples** [comment]
got few details from Rob:

from our sec audit: logging out of idp doesn’t prevent the jwt token being replayed if it was captured during logout. If clearing the idp token still done, likely would be a response to the UI. UI then would hit that URL after it's deleted the session

**Comment by wolfwolker** [comment]
Same behaviour here after enabling ui authentication with keycloak. Is there any updtates in this?

**Comment by emanuelbesliu** [comment]
**Proposal: opt-in RP-initiated (single) logout for OIDC providers**

Following up on this — the current `GET /auth/logout` handler (`server/route/auth.go`) only clears the local `refresh`/`session_start`/`user*` cookies and 303-redirects to `/`. The IdP session stays fully active, so hitting `/auth/sso` immediately re-authenticates with no prompt. This is a blocker for Entra ID / Keycloak deployments that require single logout for compliance.

I'd like to contribute this upstream. Proposed approach, designed to be **opt-in and fully backward-compatible**:

**1. Config (`AuthProvider` in `server/config/config.go`)**
```yaml
auth:
  providers:
    - label: sso
      # ...existing fields...
      endSessionEnabled: false            # default false -> no behavior change
      postLogoutRedirectUri: ""           # optional; falls back to base URL
```
Plus matching `TEMPORAL_AUTH_END_SESSION_ENABLED` / `TEMPORAL_AUTH_POST_LOGOUT_REDIRECT_URI` in `config/docker.yaml`.

**2. Provider wiring (`SetAuthRoutes`)**
- Read `end_session_endpoint` once from OIDC discovery via `provider.Claims(&claims)` (already available through the `coreos/go-oidc/v3` dependency — no new deps).
- Make the user's `id_token` retrievable at logout so it can be passed as `id_token_hint`.

**3. `logout()` handler**
- Clear local cookies as today, then — when `endSessionEnabled` — redirect to `end_session_endpoint?id_token_hint=<token>&post_logout_redirect_uri=<uri>` per [OIDC RP-Initiated Logout 1.0](https://openid.net/specs/openid-connect-rpinitiated-1_0.html), instead of `/`.

**4. Frontend (`temporalio/ui`, `auth-user.ts`)**
- Follow the server-issued redirect rather than the current local-only navigation.

**On the JWT-replay concern raised earlier in this thread** — that's exactly why I'd keep this strictly **opt-in and off by default**: it doesn't change ui-server's stateless/JWT session model, it just additionally terminates the IdP session for operators who need it. I'll document that ui-server sessions remain bearer-token based.

Before I open a PR: does this opt-in design look acceptable, and are there constraints (e.g. providers without a discoverable `end_session_endpoint`) you'd want handled a particular way? Happy to adjust.
